pkg/reconciler/service/service.go

```diff
@@ -382,5 +383,11 @@ func (c *Reconciler) reconcileRoute(ctx context.Context, service *v1alpha1.Servi
 	existing.Spec = desiredRoute.Spec
 	existing.ObjectMeta.Labels = desiredRoute.ObjectMeta.Labels
 	existing.ObjectMeta.Annotations = desiredRoute.ObjectMeta.Annotations
+
+	rc := route.Status.GetCondition(apis.ConditionReady)
+	if rc != nil && rc.Status == corev1.ConditionFalse {
+		existing.Status.ObservedGeneration = existing.Generation
```

Comment:
Changes to anything in Route's `Status` should _only_ occur in pkg/reconciler/route/route.go